Bootloader fails to build with MinGW (Could not find the program ['windres'])

I have been successfully cross-compiling the bootloader from Linux (amd64) to Windows using mingw for many past versions of pyinstaller. The last version that I can build is 4.2 (with the patch from https://github.com/pyinstaller/pyinstaller/issues/5544, which has since been merged into 4.4) - see [code here](https://github.com/spesmilo/electrum/blob/f066f25a30cf48447950171c404e0e52fb668347/contrib/build-wine/prepare-wine.sh#L100).

(There have been several independent breakages introduced since 4.2, I will try to investigate and report/fix them separately.)

-----

The 4.6 tag fails to build with the following error:

<details>
 <summary>build 1 (fail): Could not find the program ['windres']</summary>

```
'all' finished successfully (0.001s)
'distclean' finished successfully (0.001s)
Setting top to                           : /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader 
Setting out to                           : /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build 
Python Version                           : 3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0] 
Checking for 'gcc' (C compiler)          : i686-w64-mingw32-gcc 
System                                   : Assuming cross-compilation for Windows 
Checking size of pointer                 : 4 
Platform                                 : Windows-32bit detected based on compiler 
Checking for compiler flags -m32         : yes 
Checking for linker flags -m32           : yes 
Checking for program 'windres'           : not found 
Could not find the program ['windres']
(complete log in /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/config.log)
```
</details>

The relevant commit that introduced the issue is https://github.com/pyinstaller/pyinstaller/commit/207b10b98f86252f79de11567649b1c7c8aeda6d (as part of https://github.com/pyinstaller/pyinstaller/pull/6218).

<details>
 <summary>pyinstaller/bootloader/build/config.log  (for build 1)</summary>

```
# project PyInstallerBootloader (nodist) configured on Wed Nov  3 03:46:40 2021 by
# waf 2.0.20 (abi 20, python 3080af0 on linux)
# using ./waf all CC=i686-w64-mingw32-gcc CFLAGS=-static                               -Wno-dangling-else                               -Wno-error=unused-value                               -Wno-error=implicit-function-declaration                               -Wno-error=int-to-pointer-cast                               -Wno-error=stringop-truncation
#
----------------------------------------
Setting top to
/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader
----------------------------------------
Setting out to
/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build
----------------------------------------
Python Version
3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0]
----------------------------------------
Checking for 'gcc' (C compiler)
find program=['gcc', 'cc'] paths=['/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='CC' -> ['i686-w64-mingw32-gcc']
['i686-w64-mingw32-gcc', '-dM', '-E', '-']
out: #define __DBL_MIN_EXP__ (-1021)
#define __FLT32X_MAX_EXP__ 1024
#define __pentiumpro__ 1
#define __UINT_LEAST16_MAX__ 0xffff
#define __ATOMIC_ACQUIRE 2
#define __FLT128_MAX_10_EXP__ 4932
#define __FLT_MIN__ 1.17549435082228750796873653722224568e-38F
#define __GCC_IEC_559_COMPLEX 2
#define __UINT_LEAST8_TYPE__ unsigned char
#define __SIZEOF_FLOAT80__ 12
#define _WIN32 1
#define __INTMAX_C(c) c ## LL
#define __CHAR_BIT__ 8
#define __UINT8_MAX__ 0xff
#define __WINT_MAX__ 0xffff
#define __FLT32_MIN_EXP__ (-125)
#define __ORDER_LITTLE_ENDIAN__ 1234
#define __SIZE_MAX__ 0xffffffffU
#define __WCHAR_MAX__ 0xffff
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_1 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_2 1
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_4 1
#define __DBL_DENORM_MIN__ ((double)4.94065645841246544176568792868221372e-324L)
#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8 1
#define __GCC_ATOMIC_CHAR_LOCK_FREE 2
#define __GCC_IEC_559 2
#define __FLT32X_DECIMAL_DIG__ 17
#define __FLT_EVAL_METHOD__ 2
#define __FLT64_DECIMAL_DIG__ 17
#define __GCC_ATOMIC_CHAR32_T_LOCK_FREE 2
#define __UINT_FAST64_MAX__ 0xffffffffffffffffULL
#define __SIG_ATOMIC_TYPE__ int
#define __DBL_MIN_10_EXP__ (-307)
#define __FINITE_MATH_ONLY__ 0
#define __GNUC_PATCHLEVEL__ 0
#define __FLT32_HAS_DENORM__ 1
#define __UINT_FAST8_MAX__ 0xff
#define __has_include(STR) __has_include__(STR)
#define _stdcall __attribute__((__stdcall__))
#define __DEC64_MAX_EXP__ 385
#define __INT8_C(c) c
#define __INT_LEAST8_WIDTH__ 8
#define __UINT_LEAST64_MAX__ 0xffffffffffffffffULL
#define __SHRT_MAX__ 0x7fff
#define __LDBL_MAX__ 1.18973149535723176502126385303097021e+4932L
#define __FLT64X_MAX_10_EXP__ 4932
#define __UINT_LEAST8_MAX__ 0xff
#define __GCC_ATOMIC_BOOL_LOCK_FREE 2
#define __FLT128_DENORM_MIN__ 6.47517511943802511092443895822764655e-4966F128
#define __UINTMAX_TYPE__ long long unsigned int
#define __DEC32_EPSILON__ 1E-6DF
#define __FLT_EVAL_METHOD_TS_18661_3__ 2
#define __UINT32_MAX__ 0xffffffffU
#define __LDBL_MAX_EXP__ 16384
#define __FLT128_MIN_EXP__ (-16381)
#define __WINT_MIN__ 0
#define __FLT128_MIN_10_EXP__ (-4931)
#define __INT_LEAST16_WIDTH__ 16
#define __SCHAR_MAX__ 0x7f
#define __FLT128_MANT_DIG__ 113
#define __WCHAR_MIN__ 0
#define __INT64_C(c) c ## LL
#define __DBL_DIG__ 15
#define __GCC_ATOMIC_POINTER_LOCK_FREE 2
#define __FLT64X_MANT_DIG__ 64
#define __SIZEOF_INT__ 4
#define __SIZEOF_POINTER__ 4
#define __USER_LABEL_PREFIX__ _
#define __FLT64X_EPSILON__ 1.08420217248550443400745280086994171e-19F64x
#define __STDC_HOSTED__ 1
#define __WIN32 1
#define __LDBL_HAS_INFINITY__ 1
#define __FLT32_DIG__ 6
#define __FLT_EPSILON__ 1.19209289550781250000000000000000000e-7F
#define __SHRT_WIDTH__ 16
#define __LDBL_MIN__ 3.36210314311209350626267781732175260e-4932L
#define __STDC_UTF_16__ 1
#define __DEC32_MAX__ 9.999999E96DF
#define __FLT64X_DENORM_MIN__ 3.64519953188247460252840593361941982e-4951F64x
#define __MINGW32__ 1
#define __FLT32X_HAS_INFINITY__ 1
#define __INT32_MAX__ 0x7fffffff
#define __INT_WIDTH__ 32
#define __SIZEOF_LONG__ 4
#define __UINT16_C(c) c
#define __PTRDIFF_WIDTH__ 32
#define __DECIMAL_DIG__ 21
#define __FLT64_EPSILON__ 2.22044604925031308084726333618164062e-16F64
#define __INTMAX_WIDTH__ 64
#define __has_include_next(STR) __has_include_next__(STR)
#define __FLT64X_MIN_10_EXP__ (-4931)
#define __LDBL_HAS_QUIET_NAN__ 1
#define __FLT64_MANT_DIG__ 53
#define __GNUC__ 9
#define _cdecl __attribute__((__cdecl__))
#define __FLT_HAS_DENORM__ 1
#define __SIZEOF_LONG_DOUBLE__ 12
#define __BIGGEST_ALIGNMENT__ 16
#define __FLT64_MAX_10_EXP__ 308
#define __i686 1
#define __DBL_MAX__ ((double)1.79769313486231570814527423731704357e+308L)
#define _thiscall __attribute__((__thiscall__))
#define __INT_FAST32_MAX__ 0x7fffffff
#define __WINNT 1
#define __DBL_HAS_INFINITY__ 1
#define __WINNT__ 1
#define __HAVE_SPECULATION_SAFE_VALUE 1
#define __DEC32_MIN_EXP__ (-94)
#define __INTPTR_WIDTH__ 32
#define __FLT32X_HAS_DENORM__ 1
#define __INT_FAST16_TYPE__ short int
#define _fastcall __attribute__((__fastcall__))
#define __LDBL_HAS_DENORM__ 1
#define __FLT128_HAS_INFINITY__ 1
#define __DEC128_MAX__ 9.999999999999999999999999999999999E6144DL
#define __INT_LEAST32_MAX__ 0x7fffffff
#define __USING_SJLJ_EXCEPTIONS__ 1
#define __DEC32_MIN__ 1E-95DF
#define __DBL_MAX_EXP__ 1024
#define __WCHAR_WIDTH__ 16
#define __FLT32_MAX__ 3.40282346638528859811704183484516925e+38F32
#define __DEC128_EPSILON__ 1E-33DL
#define __ATOMIC_HLE_RELEASE 131072
#define __WIN32__ 1
#define __PTRDIFF_MAX__ 0x7fffffff
#define __ATOMIC_HLE_ACQUIRE 65536
#define __FLT32_HAS_QUIET_NAN__ 1
#define __LONG_LONG_MAX__ 0x7fffffffffffffffLL
#define __SIZEOF_SIZE_T__ 4
#define __FLT64X_MIN_EXP__ (-16381)
#define __SIZEOF_WINT_T__ 2
#define __LONG_LONG_WIDTH__ 64
#define __FLT32_MAX_EXP__ 128
#define __GXX_ABI_VERSION 1013
#define __FLT_MIN_EXP__ (-125)
#define __i686__ 1
#define __FLT64X_HAS_QUIET_NAN__ 1
#define __INT_FAST64_TYPE__ long long int
#define __FLT64_DENORM_MIN__ 4.94065645841246544176568792868221372e-324F64
#define __DBL_MIN__ ((double)2.22507385850720138309023271733240406e-308L)
#define __FLT32X_EPSILON__ 2.22044604925031308084726333618164062e-16F32x
#define __DECIMAL_BID_FORMAT__ 1
#define __FLT64_MIN_EXP__ (-1021)
#define __GXX_TYPEINFO_EQUALITY_INLINE 0
#define __FLT64_MIN_10_EXP__ (-307)
#define __FLT64X_DECIMAL_DIG__ 21
#define __DEC128_MIN__ 1E-6143DL
#define __REGISTER_PREFIX__ 
#define __UINT16_MAX__ 0xffff
#define __DBL_HAS_DENORM__ 1
#define __cdecl __attribute__((__cdecl__))
#define __FLT32_MIN__ 1.17549435082228750796873653722224568e-38F32
#define __UINT8_TYPE__ unsigned char
#define __NO_INLINE__ 1
#define __i386 1
#define __FLT_MANT_DIG__ 24
#define __LDBL_DECIMAL_DIG__ 21
#define __VERSION__ "9.3-win32 20200320"
#define __UINT64_C(c) c ## ULL
#define __GCC_ATOMIC_INT_LOCK_FREE 2
#define __FLT128_MAX_EXP__ 16384
#define __FLT32_MANT_DIG__ 24
#define _X86_ 1
#define __FLOAT_WORD_ORDER__ __ORDER_LITTLE_ENDIAN__
#define __FLT128_HAS_DENORM__ 1
#define __FLT128_DIG__ 33
#define __SCHAR_WIDTH__ 8
#define __INT32_C(c) c
#define __DEC64_EPSILON__ 1E-15DD
#define __ORDER_PDP_ENDIAN__ 3412
#define __DEC128_MIN_EXP__ (-6142)
#define __code_model_32__ 1
#define __FLT32_MAX_10_EXP__ 38
#define __INT_FAST32_TYPE__ int
#define __UINT_LEAST16_TYPE__ short unsigned int
#define __FLT64X_HAS_INFINITY__ 1
#define __INT16_MAX__ 0x7fff
#define __i386__ 1
#define __SIZE_TYPE__ unsigned int
#define __UINT64_MAX__ 0xffffffffffffffffULL
#define __FLT64X_DIG__ 18
#define __INT8_TYPE__ signed char
#define __GCC_ASM_FLAG_OUTPUTS__ 1
#define __FLT_RADIX__ 2
#define __INT_LEAST16_TYPE__ short int
#define __LDBL_EPSILON__ 1.08420217248550443400745280086994171e-19L
#define __UINTMAX_C(c) c ## ULL
#define __SIG_ATOMIC_MAX__ 0x7fffffff
#define __GCC_ATOMIC_WCHAR_T_LOCK_FREE 2
#define __SIZEOF_PTRDIFF_T__ 4
#define __FLT32X_MANT_DIG__ 53
#define __FLT32X_MIN_EXP__ (-1021)
#define __DEC32_SUBNORMAL_MIN__ 0.000001E-95DF
#define __pentiumpro 1
#define __MSVCRT__ 1
#define __INT_FAST16_MAX__ 0x7fff
#define __FLT64_DIG__ 15
#define __UINT_FAST32_MAX__ 0xffffffffU
#define __UINT_LEAST64_TYPE__ long long unsigned int
#define __FLT_HAS_QUIET_NAN__ 1
#define __FLT_MAX_10_EXP__ 38
#define __LONG_MAX__ 0x7fffffffL
#define __FLT64X_HAS_DENORM__ 1
#define __DEC128_SUBNORMAL_MIN__ 0.000000000000000000000000000000001E-6143DL
#define __FLT_HAS_INFINITY__ 1
#define __UINT_FAST16_TYPE__ short unsigned int
#define __DEC64_MAX__ 9.999999999999999E384DD
#define __INT_FAST32_WIDTH__ 32
#define __CHAR16_TYPE__ short unsigned int
#define __PRAGMA_REDEFINE_EXTNAME 1
#define __SIZE_WIDTH__ 32
#define __SEG_FS 1
#define __INT_LEAST16_MAX__ 0x7fff
#define __DEC64_MANT_DIG__ 16
#define __INT64_MAX__ 0x7fffffffffffffffLL
#define __UINT_LEAST32_MAX__ 0xffffffffU
#define __SEG_GS 1
#define __FLT32_DENORM_MIN__ 1.40129846432481707092372958328991613e-45F32
#define __GCC_ATOMIC_LONG_LOCK_FREE 2
#define __SIG_ATOMIC_WIDTH__ 32
#define __INT_LEAST64_TYPE__ long long int
#define __INT16_TYPE__ short int
#define __INT_LEAST8_TYPE__ signed char
#define __STDC_VERSION__ 201710L
#define __DEC32_MAX_EXP__ 97
#define __INT_FAST8_MAX__ 0x7f
#define __FLT128_MAX__ 1.18973149535723176508575932662800702e+4932F128
#define __INTPTR_MAX__ 0x7fffffff
#define __GXX_MERGED_TYPEINFO_NAMES 0
#define __FLT64_HAS_QUIET_NAN__ 1
#define __stdcall __attribute__((__stdcall__))
#define __FLT32_MIN_10_EXP__ (-37)
#define __FLT32X_DIG__ 15
#define __LDBL_MANT_DIG__ 64
#define __DBL_HAS_QUIET_NAN__ 1
#define __FLT64_HAS_INFINITY__ 1
#define __FLT64X_MAX__ 1.18973149535723176502126385303097021e+4932F64x
#define __SIG_ATOMIC_MIN__ (-__SIG_ATOMIC_MAX__ - 1)
#define __INTPTR_TYPE__ int
#define __UINT16_TYPE__ short unsigned int
#define __WCHAR_TYPE__ short unsigned int
#define __SIZEOF_FLOAT__ 4
#define __UINTPTR_MAX__ 0xffffffffU
#define __INT_FAST64_WIDTH__ 64
#define __DEC64_MIN_EXP__ (-382)
#define __FLT32_DECIMAL_DIG__ 9
#define __INT_FAST64_MAX__ 0x7fffffffffffffffLL
#define __GCC_ATOMIC_TEST_AND_SET_TRUEVAL 1
#define __FLT_DIG__ 6
#define __FLT32_HAS_INFINITY__ 1
#define __FLT64X_MAX_EXP__ 16384
#define __UINT_FAST64_TYPE__ long long unsigned int
#define __INT_MAX__ 0x7fffffff
#define WIN32 1
#define __INT64_TYPE__ long long int
#define __FLT_MAX_EXP__ 128
#define __DBL_MANT_DIG__ 53
#define __SIZEOF_FLOAT128__ 16
#define __INT_LEAST64_MAX__ 0x7fffffffffffffffLL
#define __GCC_ATOMIC_CHAR16_T_LOCK_FREE 2
#define __DEC64_MIN__ 1E-383DD
#define __WINT_TYPE__ short unsigned int
#define __UINT_LEAST32_TYPE__ unsigned int
#define __SIZEOF_SHORT__ 2
#define __LDBL_MIN_EXP__ (-16381)
#define __FLT64_MAX__ 1.79769313486231570814527423731704357e+308F64
#define __WINT_WIDTH__ 16
#define __INT_LEAST8_MAX__ 0x7f
#define __FLT32X_MAX_10_EXP__ 308
#define __LDBL_MAX_10_EXP__ 4932
#define __ATOMIC_RELAXED 0
#define __DBL_EPSILON__ ((double)2.22044604925031308084726333618164062e-16L)
#define __thiscall __attribute__((__thiscall__))
#define __FLT128_MIN__ 3.36210314311209350626267781732175260e-4932F128
#define __UINT8_C(c) c
#define __FLT64_MAX_EXP__ 1024
#define __INT_LEAST32_TYPE__ int
#define __SIZEOF_WCHAR_T__ 2
#define __UINT64_TYPE__ long long unsigned int
#define __FLT128_HAS_QUIET_NAN__ 1
#define __INT_FAST8_TYPE__ signed char
#define __fastcall __attribute__((__fastcall__))
#define __FLT64X_MIN__ 3.36210314311209350626267781732175260e-4932F64x
#define __GNUC_STDC_INLINE__ 1
#define __FLT64_HAS_DENORM__ 1
#define __FLT32_EPSILON__ 1.19209289550781250000000000000000000e-7F32
#define __DBL_DECIMAL_DIG__ 17
#define __STDC_UTF_32__ 1
#define __INT_FAST8_WIDTH__ 8
#define __DEC_EVAL_METHOD__ 2
#define __FLT32X_MAX__ 1.79769313486231570814527423731704357e+308F32x
#define __ORDER_BIG_ENDIAN__ 4321
#define __UINT32_C(c) c ## U
#define __INTMAX_MAX__ 0x7fffffffffffffffLL
#define __BYTE_ORDER__ __ORDER_LITTLE_ENDIAN__
#define WINNT 1
#define __FLT_DENORM_MIN__ 1.40129846432481707092372958328991613e-45F
#define __INT8_MAX__ 0x7f
#define __LONG_WIDTH__ 32
#define __UINT_FAST32_TYPE__ unsigned int
#define __CHAR32_TYPE__ unsigned int
#define __FLT_MAX__ 3.40282346638528859811704183484516925e+38F
#define __INT32_TYPE__ int
#define __SIZEOF_DOUBLE__ 8
#define __FLT_MIN_10_EXP__ (-37)
#define __FLT64_MIN__ 2.22507385850720138309023271733240406e-308F64
#define __INT_LEAST32_WIDTH__ 32
#define __INTMAX_TYPE__ long long int
#define i386 1
#define _INTEGRAL_MAX_BITS 64
#define __DEC128_MAX_EXP__ 6145
#define __FLT32X_HAS_QUIET_NAN__ 1
#define __ATOMIC_CONSUME 1
#define __GNUC_MINOR__ 3
#define __INT_FAST16_WIDTH__ 16
#define __UINTMAX_MAX__ 0xffffffffffffffffULL
#define __DEC32_MANT_DIG__ 7
#define __FLT32X_DENORM_MIN__ 4.94065645841246544176568792868221372e-324F32x
#define __DBL_MAX_10_EXP__ 308
#define __LDBL_DENORM_MIN__ 3.64519953188247460252840593361941982e-4951L
#define __INT16_C(c) c
#define __STDC__ 1
#define __PTRDIFF_TYPE__ int
#define __ATOMIC_SEQ_CST 5
#define __UINT32_TYPE__ unsigned int
#define __FLT32X_MIN_10_EXP__ (-307)
#define __UINTPTR_TYPE__ unsigned int
#define __DEC64_SUBNORMAL_MIN__ 0.000000000000001E-383DD
#define __DEC128_MANT_DIG__ 34
#define __LDBL_MIN_10_EXP__ (-4931)
#define __FLT128_EPSILON__ 1.92592994438723585305597794258492732e-34F128
#define __SIZEOF_LONG_LONG__ 8
#define __FLT128_DECIMAL_DIG__ 36
#define __GCC_ATOMIC_LLONG_LOCK_FREE 2
#define __FLT32X_MIN__ 2.22507385850720138309023271733240406e-308F32x
#define __LDBL_DIG__ 18
#define __FLT_DECIMAL_DIG__ 9
#define __UINT_FAST16_MAX__ 0xffff
#define __GCC_ATOMIC_SHORT_LOCK_FREE 2
#define __INT_LEAST64_WIDTH__ 64
#define __UINT_FAST8_TYPE__ unsigned char
#define __ATOMIC_ACQ_REL 4
#define __ATOMIC_RELEASE 3
#define __declspec(x) __attribute__((x))

find program=['ar'] paths=['/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='AR' -> ['/usr/bin/ar']
i686-w64-mingw32-gcc
----------------------------------------
System
Assuming cross-compilation for Windows
----------------------------------------
Checking size of pointer
==>

        int main() {
            static int test_array[1 - 2 * !(sizeof(void *) == 4)];
            test_array[0] = 0;
            return 0;
        }
<==
[1/2] Compiling [32mbuild/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_643ebb47e5c1940fdd2fcedf4da4ce14/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
4
----------------------------------------
Platform
Windows-32bit detected based on compiler
----------------------------------------
Checking for compiler flags -m32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/1] Compiling [32mbuild/.conf_check_c20c3e81115158a949ca797b35b9c29e/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_c20c3e81115158a949ca797b35b9c29e/testbuild/test.c.1.o']
yes
----------------------------------------
Checking for linker flags -m32
==>

int main(int argc, char **argv) {
	(void)argc; (void)argv;
	return 0;
}

<==
[1/2] Compiling [32mbuild/.conf_check_9e1806166015a83897e6175b88f481fa/test.c[0m

['i686-w64-mingw32-gcc', '-static', '-Wno-dangling-else', '-Wno-error=unused-value', '-Wno-error=implicit-function-declaration', '-Wno-error=int-to-pointer-cast', '-Wno-error=stringop-truncation', '-m32', '../test.c', '-c', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_9e1806166015a83897e6175b88f481fa/testbuild/test.c.1.o']
[2/2] Linking [33mbuild/.conf_check_9e1806166015a83897e6175b88f481fa/testbuild/testprog.exe[0m

['i686-w64-mingw32-gcc', '-Wl,--enable-auto-import', '-m32', 'test.c.1.o', '-o/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/.conf_check_9e1806166015a83897e6175b88f481fa/testbuild/testprog.exe', '-Wl,-Bstatic', '-Wl,-Bdynamic']
yes
----------------------------------------
Checking for program 'windres'
not found
find program=['windres'] paths=['', '/usr/local/sbin', '/usr/local/bin', '/usr/sbin', '/usr/bin', '/sbin', '/bin'] var='WINRC' -> None
from /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader: Could not find the program ['windres']
```
</details>


If I try to build the parent of that commit (namely https://github.com/pyinstaller/pyinstaller/commit/a1a7e96e656f94fbf5ae3d608b6f80b4cc2ac9fd), `windres` is correctly found (there is an unrelated error later in the build tracked at https://github.com/pyinstaller/pyinstaller/issues/6338):

<details>
 <summary>build 2: from parent of relevant commit</summary>

```
'all' finished successfully (0.001s)
'distclean' finished successfully (0.000s)
Setting top to                           : /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader 
Setting out to                           : /opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build 
Python Version                           : 3.8.10 (default, Sep 28 2021, 16:10:42) [GCC 9.3.0] 
Checking for 'gcc' (C compiler)          : i686-w64-mingw32-gcc 
System                                   : Assuming cross-compilation for Windows 
Checking size of pointer                 : 4 
Platform                                 : Windows-32bit detected based on compiler 
Checking for compiler flags -m32         : yes 
Checking for linker flags -m32           : yes 
Checking for program 'i686-w64-mingw32-windres' : /usr/bin/i686-w64-mingw32-windres 
Checking for library user32                     : yes 
Checking for library comctl32                   : yes 
Checking for library kernel32                   : yes 
Checking for library advapi32                   : yes 
Checking for library gdi32                      : yes 
Checking for library cmocka                     : not found 
Checking for function unsetenv                  : no 
Checking for function mkdtemp                   : no 
Checking for function dirname                   : yes 
Checking for function basename                  : yes 
Checking for function strndup                   : no 
Checking for function strnlen                   : yes 
Checking for program 'i686-w64-mingw32-strip'   : /usr/bin/i686-w64-mingw32-strip 
Checking for program 'strip'                    : /usr/bin/i686-w64-mingw32-strip 
'configure' finished successfully (1.304s)
'make_all' finished successfully (0.008s)
Waf: Entering directory `/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/debug'
[ 1/24] Compiling zlib/unzip.c
[ 2/24] Compiling zlib/inffast.c
[ 3/24] Compiling zlib/adler32.c
[ 4/24] Compiling zlib/ioapi.c
[ 5/24] Compiling zlib/inftrees.c
[ 6/24] Compiling zlib/crc32.c
[ 7/24] Compiling zlib/inflate.c
[ 8/24] Compiling zlib/zutil.c
[ 9/24] Compiling src/pyi_exception_dialog.c
[10/24] Compiling src/pyi_win32_utils.c
[11/24] Compiling src/pyi_splashlib.c
[12/24] Compiling src/pyi_utils.c
[13/24] Compiling src/pyi_global.c
[14/24] Linking build/debug/libstatic_zlib.a
[15/24] Compiling src/pyi_splash.c
[16/24] Compiling src/pyi_main.c
[17/24] Compiling src/pyi_python.c
[18/24] Compiling src/pyi_launch.c
[19/24] Compiling src/pyi_archive.c
[20/24] Compiling src/pyi_path.c
[21/24] Compiling src/pyi_pythonlib.c
[22/24] Compiling src/main.c
[23/24] Linking build/debug/run_d.exe
[24/24] Processing build/debug/run_d.exe
Waf: Leaving directory `/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/debug'
'build_debug' finished successfully (4.192s)
Waf: Entering directory `/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/release'
[ 1/24] Compiling zlib/unzip.c
[ 2/24] Compiling zlib/inffast.c
[ 3/24] Compiling zlib/adler32.c
[ 4/24] Compiling zlib/ioapi.c
[ 5/24] Compiling zlib/inftrees.c
[ 6/24] Compiling zlib/crc32.c
[ 7/24] Compiling zlib/inflate.c
[ 8/24] Compiling zlib/zutil.c
[ 9/24] Compiling src/pyi_exception_dialog.c
[10/24] Compiling src/pyi_splashlib.c
[11/24] Compiling src/pyi_utils.c
[12/24] Compiling src/pyi_global.c
[13/24] Compiling src/pyi_pythonlib.c
[14/24] Compiling src/pyi_archive.c
[15/24] Linking build/release/libstatic_zlib.a
[16/24] Compiling src/pyi_main.c
[17/24] Compiling src/pyi_launch.c
[18/24] Compiling src/pyi_python.c
[19/24] Compiling src/pyi_win32_utils.c
[20/24] Compiling src/pyi_path.c
[21/24] Compiling src/pyi_splash.c
[22/24] Compiling src/main.c
[23/24] Linking build/release/run.exe
[24/24] Processing build/release/run.exe
Waf: Leaving directory `/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/release'
'build_release' finished successfully (4.103s)
Waf: Entering directory `/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/debugw'
[ 1/24] Compiling zlib/unzip.c
[ 2/24] Compiling zlib/inffast.c
[ 3/24] Compiling zlib/adler32.c
[ 4/24] Compiling zlib/ioapi.c
[ 5/24] Compiling zlib/inftrees.c
[ 6/24] Compiling zlib/crc32.c
[ 7/24] Compiling zlib/inflate.c
[ 8/24] Compiling zlib/zutil.c
[ 9/24] Compiling src/pyi_exception_dialog.c
[10/24] Compiling src/pyi_splashlib.c
[11/24] Compiling src/pyi_utils.c
[12/24] Compiling src/pyi_global.c
../../src/pyi_exception_dialog.c: In function ‘_exception_dialog_initialze’:
../../src/pyi_exception_dialog.c:217:5: warning: implicit declaration of function ‘LoadIconMetric’ [-Wimplicit-function-declaration]
  217 |     LoadIconMetric(NULL, MAKEINTRESOURCEW(IDI_ERROR), LIM_LARGE, &dialog->hErrorIcon);
      |     ^~~~~~~~~~~~~~
../../src/pyi_exception_dialog.c:217:55: error: ‘LIM_LARGE’ undeclared (first use in this function); did you mean ‘SHIL_LARGE’?
  217 |     LoadIconMetric(NULL, MAKEINTRESOURCEW(IDI_ERROR), LIM_LARGE, &dialog->hErrorIcon);
      |                                                       ^~~~~~~~~
      |                                                       SHIL_LARGE
../../src/pyi_exception_dialog.c:217:55: note: each undeclared identifier is reported only once for each function it appears in

Waf: Leaving directory `/opt/wine64/drive_c/electrum/contrib/build-wine/.cache/win32/pyinstaller/bootloader/build/debugw'
Build failed
 -> task in 'OBJECTS' failed with exit status 1 (run with -v to display more information)
```
</details>

--- a/bootloader/wscript
+++ b/bootloader/wscript
@@ -414,16 +414,6 @@ def set_arch_flags(ctx):
             # TODO: verify if this option being as default might cause any side effects.
             ctx.env.append_value('LINKFLAGS', '-Wl,--large-address-aware')
 
-    # We need to pass architecture switch to the 'windres' tool.
-    if ctx.env.DEST_OS == 'win32' and ctx.env.CC_NAME != 'msvc':
-        if ctx.env.PYI_ARCH == '32bit':
-            ctx.env.WINRCFLAGS = ['--target=pe-i386']
-        else:
-            ctx.env.WINRCFLAGS = ['--target=pe-x86-64']
-        # Since WINRC config changed above, must set other options as well
-        ctx.env.WINRC_TGT_F = '-o'
-        ctx.env.WINRC_SRC_F = '-i'
-
 
 # A bare minimum zlib program. Testing zlib-dev availability requires explicitly using libz-provided symbols instead of
 # just compiling 'include <zlib.h>' with `-lz`, because while zlib's development headers may be installed, the library
@@ -504,12 +494,6 @@ def configure(ctx):
         # Do not embed manifest file when using MSVC (Visual Studio). Manifest file will be added in the phase of
         # packaging python application by PyInstaller.
         ctx.env.MSVC_MANIFEST = False
-
-        if ctx.env.CC_NAME != 'msvc':
-            # Load tool to process *.rc* files for C/C++ like icon for exe files. For msvc, waf loads this tool
-            # automatically.
-            find_program_next_to_cc(ctx, 'windres', var='WINRC')
-            ctx.load('winres')
 
     # ** C Compiler optimizations **
     # TODO Set proper optimization flags for MSVC (Visual Studio).

--- a/news/6339.bootloader.rst
+++ b/news/6339.bootloader.rst
@@ -1,0 +1,2 @@
+(Windows) Remove the check for the unused ``windres`` utility when compiling 
+with MinGW toolchain.

